api/include/opentelemetry/context/runtime_context.h

```diff
@@ -275,7 +275,14 @@ class ThreadLocalContextStorage : public RuntimeContextStorage
       Context *temp = new Context[new_capacity];
       if (base_ != nullptr)
       {
-        std::copy(base_, base_ + old_size, temp);
+        // vs2015 does not like this construct considering it unsafe:
```

Comment:
Check Visual Studio 2015 compiler like via `_MSC_VER` to separate the workaround for it?

Reply:
Does `std::copy` itself perform a bounds checking? i.e. what if new temp size - `new_capacity` is less than the old capacity? Since it's a raw pointer, I think the old code had a problem.